public/Get-VagrantEnvironment.ps1

```diff
@@ -16,30 +16,47 @@ function Get-VagrantEnvironment {
     #>
     
     [cmdletBinding()]
-    Param()
+    Param(
+        [switch]
+        $All
+    )
 
     process {
+        if ($All) {
+            $environments = vagrant global-status
         
-        $environments = vagrant global-status
-        
-        $environments = $environments[2..$($environments.Length-8)]
-
-        Foreach($environment in $environments){
+            $environments = $environments[2..$($environments.Length - 8)]
+            Foreach ($environment in $environments) {
 
-            if($environment -match "^(?<id>[\w]+)\s+(?<name>[\w\-._]+)\s+(?<provider>[\w]+)\s+(?<state>[\w]+)\s+(?<path>[\/\-\w:]+)"){
-           
-                [pscustomobject]@{
-                    Id = $matches.id
-                    Name = $matches.name
-                    Provider = $matches.provider
-                    State = $matches.state
-                    Path = $matches.path
-                }
+                if ($environment -match "^(?<id>[\w]+)\s+(?<name>[\w\-._]+)\s+(?<provider>[\w]+)\s+(?<state>[\w]+)\s+(?<path>[\/\-\w:]+)") {
+                    [pscustomobject]@{
+                        Id       = $matches.id
+                        Name     = $matches.name
+                        Provider = $matches.provider
+                        State    = $matches.state
+                        Path     = $matches.path
+                    }
             
+                }
+                
             }
+        }
+        else {
+            $environments = vagrant status
             
+            $environments = $environments[2..$($environments.Length - 4)]
+            
+            foreach($environment in $environments){
+                if ($environment -match "^(?<name>[\w\-._]+)\s+(?<state>[\w]+)\s+\((?<provider>[\w]+)\)") {

```

Comment:
This regex is subtley wrong... Two `state`s that I've seen are `not created` which won't be picked up and running which will. Further: the `name` can include non-word characters like `-`.